### PR TITLE
[overlay] Add back INSetProfileInCarIntent.defaultProfile for compatibliity

### PR DESCRIPTION
**Pull request by moiseev**
<rdar://problem/33457609>

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

File: stdlib/public/SDK/Intents/INSetProfileInCarIntent.swift [hunk 1]
```diff
@@ -128,6 +128,11 @@ extension INSetProfileInCarIntent {
         return __defaultProfile?.boolValue
     }
 
+    @available(iOS, deprecated: 11.0, message: "Please use isDefaultProfile instead")
```
Comment (jrose-apple) [inline stdlib/public/SDK/Intents/INSetProfileInCarIntent.swift]: Should these deprecations really be based on deployment target and not Swift version?

Comment (moiseev) [inline stdlib/public/SDK/Intents/INSetProfileInCarIntent.swift]: Probably not.

Comment (roeheat) [inline stdlib/public/SDK/Intents/INSetProfileInCarIntent.swift]: Yes, deprecating on swift version makes more sense.

Comment (jrose-apple) [inline stdlib/public/SDK/Intents/INSetProfileInCarIntent.swift]: That's going to apply to everything in the overlay, so maybe a separate commit!

Comment (moiseev) [inline stdlib/public/SDK/Intents/INSetProfileInCarIntent.swift]: @jrose-apple it is not that simple elsewhere in this file. Needs some extra thinking...

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Pull request merged**